classes/services/queryBuilders/PKPStatsEditorialQueryBuilder.php

```diff
@@ -387,7 +387,14 @@ protected function _getObject(): Builder
         $q = $this->_getBaseQuery();
 
         // Exclude incomplete submissions
-        $q->where('s.submission_progress', '=', 0);
+        // although the values for submission_progress column have changed
+        // (s. classes/migration/upgrade/v3_4_0/I7191_SubmissionProgressType.php)
+        // so that 0 should be ''
+        // still consider 0 because the QuickSubmitPlugin still inserts 0
```

Comment:
I think other places that make use of the submission progress (I didn't inspect how far it goes), should be updated too.

Hmm, but instead of including workarounds due to issues on non-native plugins (e.g. the "custom" heuristic, based on the QuickSubmit plugin, that we've got to determine if a submission is imported doesn't satisfy other methods/plugins), I think it's better to find a solution with migrations (`submission_progress: 0 => '', 1 => 'start'`).

- A migration for 3.5 is fine, the plugin should be fixed by then.
- The migration for 3.4, it also doable, but as an outdated plugin might generate issues again, perhaps we can do a pre-flight check "Upgrade the QuickSubmit at least to the version X.Y.Z before upgrading" 👀

Reply:
@jonasraoni, do you then suggest not to consider 0 here?
Maybe actually the QuickSubmit should implement the upgrade script that will repair this, if there are no other places in the core code that needs to be fixed...

Reply:
Yeah, just to ensure the data is normalized.